app/helpers/photo_helper.rb

```diff
@@ -0,0 +1,108 @@
+module PhotoHelper
+
+  def search_photos rover, params
+    photos = rover.photos.order(:camera_id, :id).search params, rover
+
+    if params[:page]
+      photos = photos.page(params[:page]).per params[:per_page]
+    end
+
+    photos
+  end
+
+  class InvalidSizeParameter < StandardError
+    def initialize(
+        size,
+        rover_name,
+        exception_type="custom"
+      )
+      super("Invalid size parameter '#{size}' for #{rover_name.titleize}")
+    end
+  end
+
+  def resize_photo photo, params
```

Comment:
I think what I would do here is delete the `resize_photo` method, and instead call `resize_photos([photo])` in the case of the `show` action. What are your thoughts on that?

Reply:
I think I can make that change pretty easily. The only reason I have two different methods is because I don't expect `:rover_id` to exist in `params` in the `show` action. But I can just `params.merge` that in before calling `resize_photos [photo], params`.